### PR TITLE
Update densityplot.R

### PR DESCRIPTION
**Pull request by rcannood**

### CODE REVIEW

**Comment by jacobpwagner** [comment]
@rcannood, could you provide an example or rationale for the need for this change?

**Comment by rcannood** [comment]
Hello @jacobpwagner,

Sorry for not providing a description earlier. We ran into an issue where an error was being plotted into our plots. I traced it back to this line of code; it seemed to me that the `)` was a the wrong place:

```> df <- data.frame(x = runif(10), y = runif(10, 10, 20))
> df
            x        y
1  0.14188830 15.65374
2  0.03833982 14.88725
3  0.76410898 10.45770
4  0.26343467 11.08231
5  0.80041940 14.29357
6  0.88100426 16.73579
7  0.32862713 14.54735
8  0.45252381 18.76021
9  0.50977160 19.68519
10 0.28668028 11.13584
> range(df)
[1]  0.03833982 19.68518657
> range(df)[,"x"]
Error in range(df)[, "x"] : incorrect number of dimensions
> range(df[,"x"])
[1] 0.03833982 0.88100426
```

File: R/densityplot.R [hunk 1]
```diff
@@ -563,7 +563,7 @@ panel.densityplot.flowFrame <-
           ptList<-plotType("gdensity", parm)
          
           xx <- exprs(frame)[,channel.x.name]
-          r <- unlist(range(frame)[, channel.x.name])
+          r <- unlist(range(frame[, channel.x.name]))
```
Comment (gfinak) [inline R/densityplot.R]: This should actually be:
`r <- unlist(range(frame[, channel.x.name, drop=FALSE]))`

**Comment by jacobpwagner** [comment]
Thanks @rcannood. The difference here is that in the example `df` is not a `flowFrame`, so `range` isn't dispatching to `flowCore::range()` for `flowFrame`. I didn't originally write this code, but I believe it may have been written this way to avoid unnecessary subsetting of the `flowFrame` to recalculate the range. In my testing on `flowFrame` objects I haven't been able to replicate the error, but this change makes it >3 times slower. Can you provide an example where a `flowFrame` was bringing about the error?

**Comment by gfinak** [comment]
@rcannood Your example shows a `data.frame` but the code works on a `flowFrame` which behaves differently.
```
m<-matrix(rnorm(100000),ncol=10)
colnames(m)<-LETTERS[1:10]
channel.x.name="G"
#original code
unlist(range(frame)[, channel.x.name])
[1] -3.709668  3.957575
#proposed change
unlist(range(frame[, channel.x.name]))
       G1        G2 
-3.709668  3.957575 

library(microbenchmark)
microbenchmark(unlist(range(frame[, channel.x.name])),unlist(range(frame)[, channel.x.name]))
Unit: microseconds
                                   expr      min        lq      mean    median
 unlist(range(frame[, channel.x.name])) 1930.041 2197.2630 2521.6732 2511.8360
 unlist(range(frame)[, channel.x.name])  129.124  150.7665  176.0543  175.1055
        uq      max neval cld
 2634.5260 3758.784   100   b
  194.5015  281.412   100  a 
```
The proposed change runs over ten times slower. And we can't reproduce your error, so I don't see the need to accept this change. Unless you can provide a reproducible example for the error you are getting.

**Comment by gfinak** [comment]
@rcannood  Have a look at the `reprex` package for producing a reproducible example of your bug. Paste that code into this issue, I'll re-open it and address it.

**Pull request closed without merging**